frontend/javascripts/viewer/model/sagas/update_actions.ts

```diff
@@ -373,30 +400,137 @@ export function updateVolumeTracing(
     },
   } as const;
 }
-export function updateUserBoundingBoxesInSkeletonTracing(
-  userBoundingBoxes: Array<UserBoundingBox>,
+
+export function addUserBoundingBoxInSkeletonTracingAction(
+  boundingBox: UserBoundingBox,
+  actionTracingId: string,
+) {
+  return {
+    name: "addUserBoundingBoxInSkeletonTracing",
+    value: {
+      boundingBox: convertUserBoundingBoxesFromFrontendToServer(boundingBox),
+      actionTracingId,
+    },
+  } as const;
+}
+
+export function addUserBoundingBoxInVolumeTracingAction(
+  boundingBox: UserBoundingBox,
+  actionTracingId: string,
+) {
+  return {
+    name: "addUserBoundingBoxInVolumeTracing",
+    value: {
+      boundingBox: convertUserBoundingBoxesFromFrontendToServer(boundingBox),
+      actionTracingId,
+    },
+  } as const;
+}
+
+export function deleteUserBoundingBoxInSkeletonTracingAction(
+  boundingBoxId: number,
   actionTracingId: string,
 ) {
   return {
-    name: "updateUserBoundingBoxesInSkeletonTracing",
+    name: "deleteUserBoundingBoxInSkeletonTracing",
     value: {
+      boundingBoxId,
       actionTracingId,
-      boundingBoxes: convertUserBoundingBoxesFromFrontendToServer(userBoundingBoxes),
     },
   } as const;
 }
-export function updateUserBoundingBoxesInVolumeTracing(
-  userBoundingBoxes: Array<UserBoundingBox>,
+
+export function deleteUserBoundingBoxInVolumeTracingAction(
+  boundingBoxId: number,
   actionTracingId: string,
 ) {
   return {
-    name: "updateUserBoundingBoxesInVolumeTracing",
+    name: "deleteUserBoundingBoxInVolumeTracing",
     value: {
+      boundingBoxId,
       actionTracingId,
-      boundingBoxes: convertUserBoundingBoxesFromFrontendToServer(userBoundingBoxes),
     },
   } as const;
 }
+
+function getUpdateUserBoundingBoxAction(
+  actionName: "updateUserBoundingBoxInVolumeTracing" | "updateUserBoundingBoxInSkeletonTracing",
+  boundingBoxId: number,
+  updatedProps: PartialBoundingBoxWithoutVisibility,
+  actionTracingId: string,
+) {
+  let serverBBox = EMPTY_OBJECT;
+  const { boundingBox, ...rest } = updatedProps;
+  const updatedPropKeys = Object.keys(updatedProps);
+  if (boundingBox != null) {
+    serverBBox = { boundingBox: Utils.computeBoundingBoxObjectFromBoundingBox(boundingBox) };
+  }
+  return {
+    name: actionName,
+    value: {
+      boundingBoxId,
+      actionTracingId,
+      updatedProps: { ...serverBBox, ...rest },
+      updatedPropKeys,
+    },
+  } as const;
+}
+
+export function updateUserBoundingBoxInVolumeTracingAction(
+  boundingBoxId: number,
+  updatedProps: PartialBoundingBoxWithoutVisibility,
+  actionTracingId: string,
+) {
+  return getUpdateUserBoundingBoxAction(
+    "updateUserBoundingBoxInVolumeTracing",
+    boundingBoxId,
+    updatedProps,
+    actionTracingId,
+  );
+}
+
+export function updateUserBoundingBoxInSkeletonTracingAction(
+  boundingBoxId: number,
+  updatedProps: Partial<UserBoundingBox>,
```

Comment:
should it be `PartialBoundingBoxWithoutVisibility` like in the sibling function?